backends/kv/reader.go

```diff
@@ -170,12 +178,15 @@ func (ki *Iterator) Next() bool {
 			}
 
 			var err error
-			err = utils.AppendNil(col)
+			err = utils.AppendNil(byName[name])
```

Comment:
The implementation of `utils.AppendNil` does not make sense to me
IMHO, you should use complex data types (SQL data type structs) to represent the values, this will allow full and correct support of nullable values

Reply:
After F2F discussion we have had with @talIguaz I've convinced that with current implementation of frames the idea of using a map of column indices (map[int]bool) per row is a viable solution.
I also understood that with this solution actual value of the  field that has no value (None/Null) is not important at all, i.e init value in that case could be the default value of corresponding data type.